### ISSUE TITLE
Kompiler fails on syntactic list production with nullary `symbol` attribute

### ISSUE THREAD

**Issue by tothtamas28** [runtimeverification/k#4067]
Related:
* #3996
* #4036

K version: `v6.3.23`

```k
// int-list.k
module INT-LIST
    imports INT-SYNTAX
    syntax Ints ::= List{Int, ","} [klabel(int-list), symbol]
endmodule
```

Kompile:

```
$ kompile int-list.k --syntax-module INT-LIST
```

Output:

```
[Error] Compiler: The 1-argument form of the `symbol(_)` attribute cannot be
combined with `klabel(_)`.
        Source(/home/ttoth/git/int-list.k)
        Location(4,21,4,62)
        4 |         syntax Ints ::= List{Int, ","} [klabel(int-list), symbol]
          .                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[Error] Compiler: Had 1 structural errors.
```

## How to fix it

Productions that trigger this warning are syntactic list productions with the `klabel` attribute.

1. If the list production is part of an overload, replace `klabel(foo)` by `overload(foo)`.
2. Otherwise, if the production doesn't have `symbol`, either
   1. drop `klabel(foo)` if the KAST symbol is irrelevant, or
   2. replace `klabel(foo)` by `symbol(foo)` if the KAST symbol name is relevant.
4. Otherwise, replace `klabel(foo), symbol` by `symbol(foo)`.

Since `klabel` is deprecated and the fix is straightforward, we might not want to fix this issue.

**Comment by geo2a** [comment]
We have a similar issue in one of the integration [tests](https://github.com/runtimeverification/haskell-backend/blob/0dcecfab68794c3b47d32404145dac0f9690f65d/test/overloads/simple-exps/simple-exps.k#L8) in `haskell-backend`:

```
 Error:  Compiler: The 1-argument form of the `symbol(_)` attribute cannot be
combined with `klabel(_)`.
	Source(/home/github-runner/actions-runner/_work/haskell-backend/haskell-backend/test/overloads/simple-exps/simple-exps.k)
	Location(9,19,9,55)
	9 |	  syntax Vals ::= List{Val,","}          [klabel(_,_)]
	  .	                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

The error message is the same, but the production does not have the `symbol` attribute.

**Comment by geo2a** [comment]
Ok, so perhaps this is a broader issue and some of the integration tests in [haskell-backend/test/overloads](https://github.com/runtimeverification/haskell-backend/tree/master/test/overloads) should be changed/deprecated. @tothtamas28 may I ask you to have a look at these tests and advise which do you think should we keep?

**Comment by tothtamas28** [comment]
I think you should keep all of them. Here's how to fix the K definitions:

1. For `klabel`-based overloads, replace `klabel(foo)` by `overload(foo)`.
2. For syntactic list productions with `klabel(foo)` and without `symbol`, just drop `klabel`
3. For syntactic list productions with `klabel(foo)` and `symbol`, replace `klabel(foo), symbol` by `symbol(foo)`.

Do these in order so that each overloaded list production gets the `overload`.

If KORE output is compared, tests where `klabel(foo)` is replaced by `symbol(foo)` might break. The expected output can be updated with the new KORE symbol names using search-and-replace.

**Edit**: updated the issue description with these fixes.

**Comment by Baltoli** [comment]
> Since klabel is deprecated and the fix is straightforward, we might not want to fix this issue.

Agreed; the problematic attributes will be going away anyway soon anyway and we'll have to fix this problem anyway, so I'm inclined to just let this fix get applied gradually to projects in advance of that happening. We should leave the issue open until such a time as it's resolved, though.

**Comment by geo2a** [comment]
The fixes suggested worked well, thanks @tothtamas28 !
